examples/video_multipart_create.py

```diff
@@ -2,53 +2,108 @@
-    :param local_video_path: <string> Path to media on local machine.
-    :param api_format: <string> REQUIRED Acceptable values include 'py','xml','json',and 'php'
-    :param kwargs: Arguments conforming to standards found @ https://developer.jwplayer.com/jw-platform/reference/v1/methods/videos/create.html
-    :return:
+    Configures all of the needed upload_parameters and sets up all information pertinent
+    to the video to be uploaded.
+
+    :param video_file_path: <str> the absolute path to the video file
     """
-    # Setup API client
-    jwplatform_client = jwplatform.Client(api_key, api_secret)
+    upload_parameters = dict()
+    upload_parameters['file_path'] = video_file_path
+    upload_parameters['file_size'] = os.stat(video_file_path).st_size
+    upload_parameters['file_name'] = os.path.basename(video_file_path)
 
-    # Make /videos/create API call
-    logging.info("Registering new Video-Object")
     try:
-        response = jwplatform_client.videos.create(upload_method='multipart', **kwargs)
-    except jwplatform.errors.JWPlatformError as e:
-        logging.error("Encountered an error creating a video\n{}".format(e))
-    logging.info(response)
-
-    # Construct base url for upload
-    upload_url = '{}://{}{}'.format(
-        response['link']['protocol'],
-        response['link']['address'],
-        response['link']['path']
+        # Setup API client
+        jwplatform_client = Client(JW_API_KEY, JW_API_SECRET)
+
+        # Make /videos/create API call with multipart parameter specified
+        jwplatform_video_create_response = jwplatform_client.videos.create(
+            upload_method='multipart',
+            title=upload_parameters['file_name']
+        )
+
+        # Construct base url for upload
+        upload_parameters['upload_url'] = '{}://{}{}'.format(
+            jwplatform_video_create_response['link']['protocol'],
+            jwplatform_video_create_response['link']['address'],
+            jwplatform_video_create_response['link']['path']
+        )
+        logging.info('Upload URL to be used: {}'.format(upload_parameters['upload_url']))
+
+        upload_parameters['query_parameters'] = jwplatform_video_create_response['link']['query']
+        upload_parameters['query_parameters']['api_format'] = 'json'
+        upload_parameters['headers'] = {'X-Session-ID': jwplatform_video_create_response['session_id']}
+        # The chunk offset will be updated several times during the course of the upload
+        upload_parameters['chunk_offset'] = 0
+
+    except JWPlatformError:
+        logging.error('An error occurred during the uploader setup. Check that your API keys are properly '
+                      'set up in your environment, and ensure that the video file path exists.')
+        raise
+
+    # Perform the multipart upload
+    try:
+        with open(upload_parameters['file_path'], 'rb') as file_to_upload:
+            chunk = file_to_upload.read(BYTES_TO_BUFFER)
```

Comment:
can we move this inside loop? like
`while True:`
    `chunk = file_to_upload.read(BYTES_TO_BUFFER)`
   ` if len(chunk) <= 0:`
       ` break`
   ` upload_chunk(chunk, upload_parameters)`
also we can just wrap try/exception to specific line